grails-app/services/org/pih/warehouse/receiving/ReceiptService.groovy

```diff
@@ -69,11 +69,11 @@ class ReceiptService {
      * @param shipment
      * @return
      */
-    PartialReceipt getPartialReceiptFromShipment(Shipment shipment) {
+    PartialReceipt getPartialReceiptFromShipment(Shipment shipment, User user) {
 
         PartialReceipt partialReceipt = new PartialReceipt()
         partialReceipt.shipment = shipment
-        partialReceipt.recipient = shipment.recipient
+        partialReceipt.recipient = user
```

Comment:
Can you explain why we're doing this?

Reply:
Also, there's a better way to get the current user without passing it down through a bunch of methods. 
```
def currentUser = AuthService.currentUser.get()
```

But I'd like to know why we're passing the user into these methods before I recommend this approach.

Reply:
Recipient on receipt should be the person who created partial receipt so the current user. I don't know why we used to set shipment.recipient because this field is null and in database all receipts do not have recipient.

Reply:
Ok, sounds good.